src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -417,9 +417,9 @@ void GcsServer::InitGcsActorManager(const GcsInitData &gcs_init_data) {
         gcs_placement_group_manager_->CleanPlacementGroupIfNeededWhenActorDead(actor_id);
       },
       [this](std::function<void(void)> fn, boost::posix_time::milliseconds delay) {
-        boost::asio::deadline_timer timer(main_service_);
-        timer.expires_from_now(delay);
-        timer.async_wait([fn](const boost::system::error_code &error) {
+        auto timer = std::make_shared<boost::asio::deadline_timer>(main_service_);
```

Comment:
I found this logic actually never worked before lol... the reason is the timer is GC'ed as soon as this block is exited which makes the async_wait aborted. So this logic never worked. cc @scv119 I will also see if there's other usage like this from the codebase as a follow up,.

cc @ericl actually should we just completely remove this GC actors when the job terminates? We already have logic to clean up dead actor metadata by the # of dead actors count. https://github.com/ray-project/ray/blob/073e7bc04d989607848552537f9f5ac91fa07d85/src/ray/gcs/gcs_server/gcs_actor_manager.cc#L1479

Reply:
I think I agree it makes sense to remove this logic if we already have another mechanism to remove actors, but I don't have the full context.